rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/topology/TopologyManager.java

```diff
@@ -585,19 +632,22 @@ public NodeSet select(int number, List<Node> matchedNodes) {
             NodeSet result = new NodeSet();
             for (Integer i : sortedCapacities) {
                 for (InetAddress host : hostsMap.get(i)) {
-                    List<Node> hostNodes = nodesOnHost.get(host);
-                    result.add(hostNodes.get(0));
-                    if (hostNodes.size() > 1) {
-                        List<Node> newExtraNodes = new LinkedList<>(hostNodes
-                                .subList(1, hostNodes.size()));
-                        if (result.getExtraNodes() == null) {
-                            result.setExtraNodes(new LinkedList<Node>());
+                    Set<Node> hostNodes = nodesOnHost.get(host);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The type of the "hostNodes" object should be an interface such as "Set" rather than the implementation "LinkedHashSet". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1319)